pkg/ingester/ingester.go

```diff
@@ -340,9 +357,9 @@ func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
 		return httpgrpc.Errorf(http.StatusBadRequest, "sample with timestamp %v is older than the maximum accepted age", sample.Timestamp)
 	}
 
-	if err := util.ValidateSample(sample); err != nil {
+	if err := ValidateSample(sample, i.cfg.MaxLengthLabelName, i.cfg.MaxLengthLabelValue); err != nil {
 		level.Error(util.WithContext(ctx, util.Logger)).Log("msg", "error validating sample", "err", err)
-		return nil
+		return err
```

Comment:
What difference does this change make?
Doesn't need to be in the code, but I would like the PR and/or commit message to say.

Reply:
Thanks for calling this out. This is a change I intended to revert... simple edit but with potential large impact. The implication is that returning err here will tell callers of ingester.Push that the data is invalid, vs silently dropping it. I am unsure how all callers would react here, so this is not something that should survive the PR IMO